grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -327,6 +327,14 @@ class ProductService {
             }
             or {
                 if (params.name) ilike("name", "%" + params.name.replaceAll(" ", "%") + "%")
```

Comment:
shouldn't this be removed at this point?

Reply:
probably not because this query first matches by the product name and if it does not find any then matches by synonym name

Reply:
yeah, but I feel like one of the `ifs` is redundant or maybe I misunderstand something - 
what about doing:
``` 
or { 
  if (params.name) {
    ilike("name", "%" + params.name.replaceAll(" ", "%") + "%")
    synonyms {
          and {
            ilike("name", "%" + params.name.replaceAll(" ", "%") + "%")
            eq("synonymTypeCode", SynonymTypeCode.DISPLAY_NAME)
          }
    }
  }
}

Reply:
Oh I see, I misunderstood what you meant by "removing"
I interpreted it as if you wanted me to remove the if and whats inside it  
like the `ilike("name", "%" + params.name.replaceAll(" ", "%") + "%")`

but I guess you meant that we can put all that inside one if instead of two.
Sounds good I can  do that.